app/routes/_layout+/dashboard.tsx

```diff
@@ -106,36 +108,48 @@ export const handle = {
 };
 
 export default function DashboardPage() {
+  const [completedAllChecks] = useState(false);
+  const matches = useMatches();
+  const parentRoutesData = matches.find(
+    (match) => match.id === "routes/_layout+/_layout"
+  )?.data;
+  const { skipOnboardingChecklist } = parentRoutesData;
```

Comment:
No reason for this to be in the parent route. Should be handled within the dashboard route.